docs/rfc/rfc-3.adoc

```diff
@@ -0,0 +1,119 @@
+:toc: macro
+
+= RFC 3: Rewards
+
+:icons: font
+:numbered:
+toc::[]
+
+== Overview
+
+When various actions are taken in the tBTC system,
+some TBTC tokens are collected as fees.
+These tokens may be sold for T tokens
+which are then distributed as rewards to various participants,
+or used as rewards directly.
+
+Operators are rewarded for being present and available in the sortition pool,
+and for performing various required transactions on the Ethereum blockchain
+as a member of an active wallet.
+In addition, any Ethereum account may be rewarded
+for performing various necessary maintenance tasks.
+
+=== Sortition pool rewards
+
+All operators present in the sortition pool
+accumulate rewards while they remain in the pool.
+For a simple solution,
+any rewards paid to the sortition pool
+are immediately divided among the currently present operators
+in proportion to their current weight in the pool.
+This requires that joining and leaving the sortition pool quickly is prevented
+so that operators can't pop in for rewards and leave afterwards,
+but requires very little bookkeeping or expensive calculations.
+
+The pool maintains a single global variable for rewards:
+`accumulatedWeightReward`.
+In addition, the total weight of the sortition pool `poolWeight` can be queried.
+Whenever `r` tokens are paid to the pool,
+the accumulator is increased: `accumulatedRewardPerWeight += r / poolWeight`.
+
+Each operator maintains 2 variables for their rewards:
+`weightRewardCollected` and `rewardAvailable`.
+In addition, each operator's `weight` can be queried.
+Whenever an operator's weight changes from `oldWeight` to `newWeight`,
+the rewards earned with the old weight are collected:
+`rewardAvailable += (accumulatedWeightReward - weightRewardCollected) * oldWeight`,
+`weightRewardCollected = accumulatedWeightReward`
+and `weight = newWeight`.
+
+At any time, an operator can withdraw their collected rewards,
+transferring `rewardAvailable` T tokens to them,
+and setting their `rewardAvailable = 0`.
```

Comment:
In [RFC 1](https://github.com/keep-network/tbtc-v2/blob/main/docs/rfc/rfc-1.adoc#continuous-fees) we say that:

> The system is able to collect fees in the form of tBTC when the BTC is deposited and when it is redeemed. Rather than paying operators at those two instances we would prefer to pay them continuously. Since we do not have the tBTC to pay them with until we collect the redemption fee (and can’t mint it and maintain the peg), we must turn to another token. Instead, the system pays operators in T tokens from a treasury funded by the T-token DAO and whenever it collects tBTC fees it uses those fees to buy T tokens back for its treasury.

Can we explain in this RFC how 

> whenever it collects tBTC fees it uses those fees to buy T tokens back for its treasury

should work?